doc/help/kdb-gen-highlevel.md

```diff
@@ -0,0 +1,101 @@
+Elektra handle will fail.
+
+The `type` metakey may only have one of the following values: `enum`, `string`, `boolean`, `char`, `octet`, `short`,
+`unsigned_short`, `long`, `unsigned_long`, `long_long`, `unsigned_long_long`, `float`, `double`, `long_double`,
+`struct`, `struct_ref` and `discriminator`. Most of these values correspond to the values supported by the high-level API,
+the remaining values (`enum`, `struct`, `struct_ref`, `discriminator`) are treated specially and are part of advanced concepts.
+For more information on these concepts take a look at [elektra-highlevel-gen(7)](elektra-highlevel-gen.md). If one of the
+advanced `type` values is used, you should also set `check/type = any`; otherwise the `type` plugin may produce an error.
+
+The template produces two output files: `<outputName>.c` and `<outputName>.h`. The `.c` file only contains implementations,
+therefore its precise content will not be documented.
+
+The header (`.h`) file contains the following parts:
+
+1. Generated `enum`s and `struct`s
+2. Declarations for generated accessor functions
+3. Tags for accessing keys
+4. `static inline` accessor functions for all tags
+5. Declarations of initialization functions
+6. Convenience accessor macros
+
+Anything else that may be part of the header file is not considered public API and may be subject to change at any point in time.
+There is also no guarantee of full compatibility between Elektra version for the documented parts of the header, however,
+we try to have as little breaking changes as possible for the six parts listed above.
+
+For detailed information about the contents of the header file see [elektra-highlevel-gen(7)](elektra-highlevel-gen.md).
+
+## PARAMETERS
+
+- `initFn`:
+  Changes the name of the initialization function (default: `loadConfiguration`)
+- `helpFn`:
+  Changes the name of the function that prints the generated help message (default: `printHelpMessage`)
+- `specloadFn`:
+  Changes the name of the function that checks for "specload mode" (default: `specloadCheck`)
+- `tagPrefix`:
+  Changes the prefix of the generated tags (default: `ELEKTRA_TAG_`)
+- `enumConv`:
+  Switches how enum conversion should be done; allowed values: `default` (default), `trie`, `strcmp`
+  - `strcmp`: uses a simple series of `if (strcmp(*, *) == 0)` to convert strings into enums
+  - `trie`: constructs a character based trie to convert strings into enums
+  - `default`: uses a `trie` up to a depth of 2, then switches to `strcmp`
+- `headers`:
+  Comma-separated (`,`) list of additional header files to include. For each of the listed headers we will generate an `#include "*"`
+  statement
+- `genSetters`:
+  Switches whether setters should be generated at all; allowed values: `true` (default), `false`
+- `specLocation`:
+  Changes where the processed specification will be stored; allowed values: `embedded` (default), `external`.
```

Comment:
This is not clear. What is a "processed specification"?

Reply:
I added an explanation to `elektra-highlevel-gen.md`. The description here is intentionally incomplete, so that people read the full documentation and don't make assumptions based on the short descriptions.